apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageFinder.java

```diff
@@ -0,0 +1,200 @@
+								PageConstants.SPE_QUESTION_COMPONENT, minSubSectionOrder, maxSubSectionOrder);
+						List<WaUiMetadata> questions = mergeOrderedQuetionLists(generalQuestions, specificQuestions);
+						List<PageDetailResponse.PageQuestion> resultQuestions = buildQuestions(questions);
+						PageDetailResponse.PageSubSection aSubSection = new PageDetailResponse.PageSubSection(
+								SectionSubSections.get(k).getId(), SectionSubSections.get(k).getQuestionLabel(),
+								SectionSubSections.get(k).getDisplayInd(), resultQuestions);
+						subSections.add(aSubSection);
+					} // End SubSection building
+
+					PageDetailResponse.PageSection aSection = new PageDetailResponse.PageSection(
+							tabSections.get(j).getId(), tabSections.get(j).getQuestionLabel(),
+							tabSections.get(j).getDisplayInd(), subSections);
+					sections.add(aSection);
+				} // EndSection building
+				PageDetailResponse.PageTab aTab = new PageDetailResponse.PageTab(pageTabs.get(i).getId(),
+						pageTabs.get(i).getQuestionLabel(), pageTabs.get(i).getDisplayInd(), sections);
+				tabs.add(aTab);
+			} // End tab
+			List<PagedDetail.PageRule> rules = getPageRules(page.get().getId());
+			PageDetailResponse.PagedDetail response = new PageDetailResponse.PagedDetail(page.get().getId(),
+					page.get().getTemplateNm(), page.get().getDescTxt(), tabs, rules);
+			return response;
+		} else {
+			throw new PageNotFoundException(PageConstants.PAGE_NOT_EXISTS + pageId);
+		}
+	}
+	
+	public List<WaUiMetadata> getChildComponents(WaTemplate page, Long nbsUiComponentUid, Integer orderNmbrMin,
+			Integer orderNmbrMax) {
+		return waUiMetadatumRepository.findOrderedChildComponents(page.getId(), nbsUiComponentUid, orderNmbrMin,
+				orderNmbrMax);
+	}
+	
+
+	public List<WaUiMetadata> mergeOrderedQuetionLists(List<WaUiMetadata> questions1, List<WaUiMetadata> questions2) {
+		if(questions1.isEmpty() && !questions2.isEmpty()) {
+			return questions2;
+		}
+		if(!questions1.isEmpty() && questions2.isEmpty()) {
+			return questions1;
+		}
+		if(questions1.isEmpty() && questions2.isEmpty()) {
+			return new ArrayList<>();
+		}
+		Integer qOneLastItemOrder = questions1.get(questions1.size() - 1).getOrderNbr();
+		Integer qTwoLastItemOrder = questions2.get(questions2.size() - 1).getOrderNbr();
+		if (qOneLastItemOrder.compareTo(qTwoLastItemOrder) > 0) {
+			questions2.addAll(questions1);
+			return questions2;
+		} else {
+			questions1.addAll(questions2);
+			return questions1;
+		}
+	}
+
+
+
+	public List<PagedDetail.PageRule> getPageRules(Long pageId) {
+		List<PagedDetail.PageRule> rules = new ArrayList<>();
+		List<WaRuleMetadata> rawRules = waRuleMetaDataRepository.findByWaTemplateUid(pageId);
+		for (WaRuleMetadata rawRule : rawRules) {
+			PagedDetail.PageRule rule = new PagedDetail.PageRule(rawRule.getId(), pageId, rawRule.getLogic(),
+					rawRule.getSourceValues(), rawRule.getRuleCd(), rawRule.getRuleDescText(), rawRule.getTargetType());
```

Comment:
The rules being returned currently look like:
```JSON
{
      "id": 7510,
      "pageId": 1000325,
      "logic": "=",
      "values": "Yes",
      "function": "Unhide",
      "sourceField": "unhide date of last IG dose if patient has received IG",
      "targetField": "QUESTION"
    }
```

I think we want to grab the `target_question_identifier` column for the `targetField` and `source_question_identifier` for the `sourceField`

Reply:
sure will update.

Reply:
@mpeels  updated